### PR TITLE
fix(QRadio): doesn't work normally when value is not proxy object

### LINKED ISSUE #13457: QOptionGroup with objects as values for the options does not reflect the selected option in UI
### What happened?

Using a default (radio-button) q-option-group with objects a as values for the options the modelValue is changed correctly on selection but the UI does not reflect the state.

### What did you expect to happen?

The radio button of the current selection should be filled (reflect the selection).

### Reproduction URL

https://codepen.io/suchwerk123/pen/WNMjNva?editors=101

### How to reproduce?

1. Make a selection
2. See that the model value changed
3. The radio button of the selected option still is not filled

### Flavour

Quasar CLI with Webpack (@quasar/cli | @quasar/app-webpack)

### Areas

Components (quasar)

### Platforms/Browsers

_No response_

### Quasar info output

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by volvachev**
Fixes #13457

**What kind of change does this PR introduce?** 

- [x] Bugfix
- [ ] Feature
- [ ] Documentation
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch (or `v[X]` branch)
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on an Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs)  or explained in the PR's description.

### CODE REVIEW

File: ui/src/components/radio/QRadio.js [hunk 1]
```diff
@@ -64,7 +64,11 @@ export default createComponent({
     const rootRef = ref(null)
     const { refocusTargetEl, refocusTarget } = useRefocusTarget(props, rootRef)
 
-    const isTrue = computed(() => props.modelValue === props.val)
+    const isTrue = computed(() => {
```
Comment (pdanpdan) [inline ui/src/components/radio/QRadio.js]: You don't need to check for `isProxy`, you can directly check `toRaw(props.modelValue) === toRaw(props.val)`

**Review by pdanpdan — CHANGES_REQUESTED** [review]

File: ui/src/components/radio/QRadio.js (hunk 1 above)
Comment (volvachev) [reply ui/src/components/radio/QRadio.js]: Yeah, sure

**Review by pdanpdan — APPROVED** [review]

**Comment by pdanpdan** [comment]
Can you please also check in `use-checkbox.js` - the branch for QCheckbox/QToggle?

It needs changing in the `index`/`isTrue`/`isFalse` computeds

```
  const index = computed(() => (
    modelIsArray.value === true
      ? props.modelValue.findIndex(opt => toRaw(opt) === toRaw(props.val))
      : -1
  ))

  const isTrue = computed(() => {
    return modelIsArray.value === true
      ? index.value > -1
      : toRaw(props.modelValue) === toRaw(props.trueValue)
  })

  const isFalse = computed(() => (
    modelIsArray.value === true
      ? index.value === -1
      : toRaw(props.modelValue) === toRaw(props.falseValue)
  ))
```

**Comment by volvachev** [comment]
> Can you please also check in `use-checkbox.js` - the branch for QCheckbox/QToggle?

Hi, you are right, same problem.

**Comment by pdanpdan** [comment]
Good job from my pov.
Now I think we should wait for Razvan to decide on what we do, because this problem might hit anywhere in the framework

**Comment by rstoenescu** [comment]
This will go into 2.7.1

**Comment by rstoenescu** [comment]
Thanks for contributing!

**Comment by suchwerk** [comment]
Yes, thank you all! For the initial workaround and fix!

**Pull request merged**